go.mod

```diff
@@ -4,23 +4,22 @@ go 1.15
 
 require (
 	github.com/metal3-io/baremetal-operator v0.0.0-00010101000000-000000000000
-	github.com/onsi/gomega v1.10.2
-	github.com/openshift/machine-api-operator v0.2.1-0.20210104142355-8e6ae0acdfcf
+	github.com/openshift/machine-api-operator v0.2.1-0.20210505133115-b7ef098180db
 	github.com/pkg/errors v0.9.1
 	github.com/stretchr/testify v1.6.1
-	golang.org/x/net v0.0.0-20201110031124-69a78807bb2b
-	k8s.io/api v0.20.0
-	k8s.io/apimachinery v0.20.0
+	golang.org/x/net v0.0.0-20210224082022-3d97a244fca7
+	k8s.io/api v0.21.0
+	k8s.io/apimachinery v0.21.0
 	k8s.io/client-go v12.0.0+incompatible
-	k8s.io/klog/v2 v2.4.0
-	sigs.k8s.io/controller-runtime v0.7.0
+	k8s.io/klog/v2 v2.8.0
+	sigs.k8s.io/controller-runtime v0.9.0-alpha.1.0.20210413130450-7ef2da0bc161
```

Comment:
I'm not sure what the motivator is to use a pre-release of 0.9, but could we at least use the latest beta.1? [available releases](https://github.com/kubernetes-sigs/controller-runtime/tags)

Reply:
this was pulled in when i updated the controller-runtime at the mao too. there was an api change which forced us to move to the 0.9.0 release, but there was only alpha available when i started this dependency update. i can update it here, but it will be different than mao and the other providers too.

Reply:
There were [a lot of changes](https://github.com/kubernetes-sigs/controller-runtime/compare/v0.9.0-alpha.1...v0.9.0-beta.1), so it seems like a good idea to go with the beta here, and in the others if possible. Among the changes is c-r's switch from k8s `v0.21.0-beta.1` to `v0.21.0`, which seems potentially important.

Reply:
oof, ok. this is gonna be a lot of updates lol